Source/Core/Gen4/Encounters4.cpp

```diff
@@ -224,6 +244,32 @@ namespace Encounters4
                 }
             }
 
+            if (encounter == Encounter::Headbutt)
+            {
+                int offset = 0;
+
+                for (int i = 0; i < 59; i++)
```

Comment:
Not a fan of this approach. It doesn't match how the rest of the tables are handled.

Would rather have
```
for (size_t offset = 0; offset < size; ) {
// Handle table

offset += (num based on special tree);
}
```

Reply:
> Not a fan of this approach. It doesn't match how the rest of the tables are handled.
> 
> Would rather have
> 
> ```
> for (size_t offset = 0; offset < size; ) {
> // Handle table
> 
> offset += (num based on special tree);
> }
> ```

done